src/use-auth.js

```diff
@@ -0,0 +1,86 @@
+import React, {
+  useState,
+  useContext,
+  createContext,
+} from 'react';
+import PropTypes from 'prop-types';
+import { useQuery, useMutation, gql } from '@apollo/client';
+
+const REGISTER = gql`
+  mutation RegisterMutation(
+    $username: String!
+    $firstName: String!
+    $lastName: String!
+    $password: String!
+  ) {
+    createUser(
+      username: $username
+      firstName: $firstName
+      lastName: $lastName
+      password: $password
+      userRoleId: 3
+    ) {
+      id
+    }
+  }
+`;
+const QUERY_USERS = gql`
+{
+  users {
+    username
+    id
+    password
+  }
+}
+`;
+
+const authContext = createContext();
+
+const useAuth = () => useContext(authContext);
+
+function useProvideAuth() {
```

Comment:
Do you think you could use this hook anywhere else other than `AuthProvider`? if not then we can just move all the code here to the main provider and rename this file to `src/services/AuthProvider.js`

Reply:
I used the design pattern shown here: https://usehooks.com/useAuth/
If there was a reason for making a separate hook and provider in that example, I'm not sure what it was. If it was only needed there because it's storing the user state on the server side, which we aren't doing here, then I can change it.

Reply:
If you feel it's best to leave those methods in that hook I'm fine with that, we can always refactor it later if we need to.
Let's just rename the provider and context to what's more commonly used(I left some suggestions about that), here is an article that could be helpful about how to restructure that layer of the code: https://kentcdodds.com/blog/application-state-management-with-react
Feel free to go with what makes the most sense to you, we can always improve it once you finish all the other pieces. If you want to have a call to discuss it more in depth, just let me know 👍